shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,236 @@
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = -1;
+    if !all_transactions.is_empty() {
+        old_seq_num = parse_txn_for_seq_num(
+            all_transactions
+                .last()
+                .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+        )?
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp =
+                get_account_transactions_response(&client, address, &network, old_seq_num + 1, 100)
+                    .await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let txn_array = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .to_vec();
+
+            // checking if there are transactions
+            if txn_array.is_empty() {
+                continue;
+            }
+            let last_txn_seq_num = parse_txn_for_seq_num(
+                txn_array
+                    .last()
+                    .ok_or_else(|| anyhow!("Couldn't get last transaction"))?,
+            )?;
+            write_out_txns(txn_array, last_txn_seq_num, old_seq_num, &mut io::stdout())?;
+            old_seq_num = last_txn_seq_num;
+        }
+    }
+    Ok(())
+}
+
+fn write_out_txns<W: Write>(
+    all_transactions: Vec<Value>,
+    last_txn_seq_num: i64,
+    old_seq_num: i64,
+    mut stdout: W,
+) -> Result<()> {
+    if last_txn_seq_num > old_seq_num {
+        for txn in all_transactions.iter() {
+            write_into(&mut stdout, txn)?;
+            println!();
+            println!();
+        }
+    }
+    Ok(())
+}
+
+pub async fn get_account_transactions_response(
+    client: &Client,
+    address: AccountAddress,
+    network: &Url,
+    start: i64,
+    limit: u64,
+) -> Result<Response> {
+    let path = network.join(format!("accounts/{}/transactions", address).as_str())?;
+    Ok(client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", limit.to_string().as_str())])
+        .send()
+        .await?)
+}
+
+pub async fn get_account_sequence_number(
+    client: &Client,
+    network: &Url,
+    address: AccountAddress,
+) -> Result<u64> {
+    let path =
+        network.join(format!("accounts/{}/resources/", address.to_hex_literal()).as_str())?;
+    let resp = client.get(path.as_str()).send().await?;
+    let json: Vec<Value> = serde_json::from_str(resp.text().await?.as_str())?;
+    parse_json_for_account_seq_num(json)
+}
+
+pub fn parse_txn_for_seq_num(last_txn: &Value) -> Result<i64> {
+    Ok(last_txn["sequence_number"]
+        .to_string()
+        .replace('"', "")
+        .parse::<i64>()?)
+}
+
+pub fn parse_json_for_account_seq_num(json_objects: Vec<Value>) -> Result<u64> {
```

Comment:
None of these methods need to be `pub`. IIUC the only method that needs to be public is `handle`. Let's return them to private since you can test them via `use super::*`.

This includes:
1. parse_json_for_account_seq_num
2. parse_txn_for_seq_num
3. get_account_sequence_number
4. get_account_transactions_response